src/main/kotlin/no/nav/personbruker/dittnav/api/common/apiUtil.kt

```diff
@@ -2,24 +2,17 @@ package no.nav.personbruker.dittnav.api.common
 
 import io.ktor.application.ApplicationCall
 import io.ktor.application.call
-import io.ktor.http.HttpHeaders
+import io.ktor.auth.authentication
 import io.ktor.util.pipeline.PipelineContext
+import no.nav.security.token.support.ktor.OIDCValidationContextPrincipal
 
-fun PipelineContext<Unit, ApplicationCall>.extractTokenFromRequest(): String {
-    var authToken = getTokenFromHeader()
-    if (authToken == null) {
-        authToken = getTokenFromCookie()
-    }
-    if (authToken == null) {
+fun PipelineContext<Unit, ApplicationCall>.hentInnloggetBruker(): InnloggetBruker {
+
+    val token = call.authentication.principal<OIDCValidationContextPrincipal>()?.context?.firstValidToken
+
+    if (token == null) {
         throw Exception("Det ble ikke funnet noe token. Dette skal ikke kunne skje.")
+    } else {
+        return InnloggetBruker(token.get())
     }
```

Comment:
Kunne denne logikken blitt lagt inn i en konstruktør for InnloggetBruker? Slik at det finnes en alternativ konstruktør som tar inn en call.authentication.principal<OIDCValidationContextPrincipal>(), og kaster feil hvis det ikke finnes et token.
Da blir koden i hentInnloggetBruker enklere å lese, og man kan i tillegg få testet tilfeller hvor context-en er null og at det ikke finnes et gyldig token.